Lib/test/test_builtin.py

```diff
@@ -358,6 +361,32 @@ def f(): """doc"""
                 rv = ns['f']()
                 self.assertEqual(rv, tuple(expected))
 
+    def test_compile_top_level_await(self):
+        for mode in ('single', 'exec'):
+            for num,code_sample in enumerate(['''await sleep(0)''',
+                                '''async for i in range(10):
+                                       print(i)''',
+                                '''async with asyncio.Lock() as l:
+                                       pass''']):
+                source = dedent(code_sample)
+                with self.assertRaises(SyntaxError,
+                                  msg='source={!r} mode={!r})'.format(source, mode)):
```

Comment:
Align msg with `SyntaxError` ? PEP 8 enforces 80 character limit per line but I will leave it to reviewer on the same.

Reply:
I've done that and used `itertools.product` to remove one level of indent. I'm still slightly over.

Reply:
Perhaps have `msg` as a variable outside of context manager.

```python
msg = 'source={!r} mode={!r})'.format(source, mode)

with self.assertRaises(SyntaxError, msg):
    compile(source, '?' , mode)
```

or even better master has f-string debugging merged into master where `f"{source=}"` expands into `f"source={source!r}"`. So you can rebase onto master and write below : 

```python
with self.assertRaises(SyntaxError, msg=f'{source=} {mode=})'): # 75 characters
    compile(source, '?' , mode)
```

Reply:
Yep good idea. Done that and rebased on master.